homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -42,6 +44,8 @@
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Optional(CONF_IP_ADDRESS, default=CONST_DEFAULT_HOST): cv.string,
+        vol.Optional(CONF_USERNAME, default=""): cv.string,
```

Comment:
Don't set a default here. Instead use `dict.get` to access the config dict so that a default `None` is used.

Reply:
@MartinHjelmare The [development docs](https://developers.home-assistant.io/docs/en/development_validation.html) say to use an empty string

> Default values for optional configuration keys need to be valid values. Don't use a default which is None like vol.Optional(CONF_SOMETHING, default=None): cv.string, set the default to default='' if required.

I'll go ahead and use `dict.get` instead, but should the online docs be updated?

Reply:
The docs talk about a case where we need a default. We don't need a default here.

Reply:
Nevermind, it looks like the library uses empty strings as defaults. 🤷‍♂️

Reply:
Although the empty string default in the library is only for the password. The username has `"envoy"` as default, which seems to be the old default too before changing the library. So if we pass an empty string here, it'll be a breaking change, right?

Should we pass `"envoy" instead?

Reply:
@MartinHjelmare I changed the default for the username to `envoy`.  I want to avoid breaking changes as much as possible. :grinning: